src/pwa-buildpack/reference/magento-resolver/index.md

```diff
@@ -0,0 +1,62 @@
+---
+title: MagentoResolver
+---
+
+An adapter that configures Webpack to resolve assets using Magento PWA conventions.
+
+This module generates a configuration object used in the [`resolve`] property of a Webpack config.
+The configuration object tells Webpack how to traverse the filesystem structure for assets required in source files.
+
+Currently, MagentoResolver does very little, but it's likely that the Magento development environment will require custom resolution rules in the future; this utility sets the precedent of the API for delivering those rules.
+
+## API
+
+### `configure(options)`
+
+**Parameters:**
+
+* `options: ResolverOptions`
+    * `options.path: object`(**required**) - Local absolute paths to theme folders
```

Comment:
Is this copied verbatim? It's a little confusing looking, and slightly inaccurate. It suggests the following:
```js
MagentoResolver.configure({
  path: {},
  root: '/some/path'
});
```

Which isn't accurate.

I think the missing piece is that we need a general "project locations type", maybe expressed with JSDoc, called `LocalProjectLocations`. A `LocalProjectLocation`'s required fields would be:

 - `root: string` -  the absolute path of the project's root directory on the working filesystem
 - `output: string` - the directory where Webpack should output its built assets. If relative, resolves from `root`.
 - `assets: string` -  the directory where any assets not explicitly built by Webpack can be found.

So then we can express that `MagentoResolver.configure()` takes a `LocalProjectLocation` as its one argument.

And then `PWADevServer` and `ServiceWorkerPlugin` and  also takes a `LocalProjectLocation` subclass as its `paths` parameter.

I know this is skirting close to TypeScript and the like, but I'm just trying to keep the configurations organized. Can I sit with you and try to figure out how to express these shared types?